catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -73,8 +123,71 @@ public Metacard transform(InputStream input, String id)
         }
     }
 
+    private Metacard transformWithExtractors(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        try (FileBackedOutputStream fbos = new FileBackedOutputStream(1000000)) {
+            try {
+                IOUtils.copy(input, fbos);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            String plainText = null;
+            try (InputStream isCopy = fbos.asByteSource()
+                    .openStream()) {
+                Parser parser = new AutoDetectParser();
+                ContentHandler contentHandler = new ToTextContentHandler();
+                TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser,
+                        contentHandler);
+                tikaMetadataExtractor.parseMetadata(isCopy, new ParseContext());
+                plainText = contentHandler.toString();
+            } catch (CatalogTransformerException e) {
+                LOGGER.warn("Cannot extract metadata from pdf", e);
+            }
+
+            try (InputStream isCopy = fbos.asByteSource()
+                    .openStream();
+                    PDDocument pdfDocument = PDDocument.load(isCopy)) {
+
+                return transformPdf(id, pdfDocument, plainText);
+            } catch (InvalidPasswordException e) {
+                LOGGER.warn("Cannot transform encrypted pdf", e);
+                return initializeMetacard(id);
+            }
+        }
+    }
+
     private MetacardImpl initializeMetacard(String id) {
-        MetacardImpl metacard = new MetacardImpl();
+        return initializeMetacard(id, null);
+    }
+
+    private MetacardImpl initializeMetacard(String id, String contentInput) {
+        MetacardImpl metacard;
+
+        if (StringUtils.isNotBlank(contentInput)) {
+            Set<AttributeDescriptor> attributes = contentMetadataExtractors.values()
+                    .stream()
+                    .map(ContentMetadataExtractor::getMetacardAttributes)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+
+            // TODO RAP 14 Jun 16: Might need to add service method to extract
+            // name/id type info from the extractor. Those would then be
+            // concatted together to form a name for the metacardtype.
+            String typeName = contentMetadataExtractors.values()
```

Comment:
This could easily result in some fairly funky metacard type names. I would recommend adding `Describable` to the metadata extractor interface. You can still fall back to the class name if an implementer `noop`s out.